doc/qos/SONiC_ROCEv2_support.md

```diff
@@ -0,0 +1,574 @@
+# Feature Name
+RDMA (Remote Direct Memory Access) over Converged Ethernet (ROCEv2) support
+
+# High Level Design Document
+
+#### Rev 0.2
+# Table of Contents
+
+  * [List of Tables](#list-of-tables)
+  * [Revision](#revision)
+  * [About This Manual](#about-this-manual)
+  * [Scope](#scope)
+  * [Definition/Abbreviation](#definitionabbreviation)
+
+# List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+# Revision
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 | 06/17/2021 |   Venkatesan Mahalingam         | Initial version                   |
+| 0.2 | 07/16/2021 |   Venkatesan Mahalingam         | Addressed review comments                   |
+
+
+# About this Manual
+
+This document introduces the support of ROCEv2 in mgmt-framework,
+this provides the North Bound Interface (NBI) interfaces i.e REST/gNMI/KLISH for the ROCEv2 configurations.
+
+# Scope
+
+This document covers the following,
+
+1) NBI interface to initialize QoS buffers for PFC \
+2) NBI interface to create/delete buffer pool \
+3) NBI interface to create/delete buffer profile \
+4) NBI interface to map/unmap priority group to buffer profile \
+5) NBI interface to set/unset PORT table for default buffer profile creation
+   based on cable length and speed
+6) Unit Testcases
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+| **Term**                 | **Meaning**                         |
+|--------------------------|-------------------------------------|
+| RDMA                     | Remote Direct Memory Access         |
+|          ROCEv2            |  RDMA over Converged Ethernet version 2           |
+|          PFC            |  Priority Flow Control|
+
+
+# 1 Feature Overview
+
+ROCE provides the ability to transport Infiniband (IB) protocol traffic using RDMA over lossless converged Ethernet

```

Comment:
Is there any way to generate IB traffic for functional validation? Request to provide more details on generation of such traffic items.

Reply:
As part of the feature testing, we're simulating traffic congestion using IXIA to test PFC, I think, system test scope is to verify the lossless IB traffic from server/storage.